src/lib.rs

```diff
@@ -201,6 +201,20 @@ impl DeviceHandle {
         self.device_type
     }
 
+    /// The [`HidDevice`] for the device.
+    #[must_use]
+    pub fn hid_device(&self) -> &HidDevice {
+        &self.hid_device
+    }
+
+    /// Returns the serial number of the device.
+    pub fn serial_number(&self) -> DeviceResult<Option<String>> {
```

Comment:
`hidapi` returns `&str`, while this methods returns a `String`. Not sure if this matters & I'm not sure on how to return a pointer when we don't store the serial number on the `DeviceHandle`.

Reply:
I did some reading about `String` vs `&str`, and I think this is the right approach.